server/mrt_converter/mrt_converter.go

```diff
@@ -0,0 +1,95 @@
+package converter
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/osrg/gobgp/pkg/packet/bgp"
+	"github.com/osrg/gobgp/pkg/packet/mrt"
+)
+
+type update struct {
+	Collector string
+	SeenAt    time.Time
+	PeerAS    uint32
+
+	// Data inside BGP updates.
+	Announced  []*bgp.IPAddrPrefix
+	Withdrawn  []*bgp.IPAddrPrefix
+	Attributes map[bgp.BGPAttrType]bgp.PathAttributeInterface
+}
+
+// extractCollector extracts the collector name from the input file path.
+func extractCollector(filename string) (string, error) {
+	if filename == "" {
+		return "", fmt.Errorf("empty file path")
+	}
+	if filename[0] != '/' {
+		return "", fmt.Errorf("path must starts with '/'")
+	}
+	if !strings.Contains(filename, "bgpdata") {
+		return "", fmt.Errorf("file %s is not a valid RouteViews archive path", filename)
+	}
+
+	dirs := strings.Split(filename, "/")
+	if dirs[1] == "bgpdata" {
+		return "route-views2", nil
+	}
+	return dirs[1], nil
+}
+
+func columnizeAttrs(attrs []bgp.PathAttributeInterface) map[bgp.BGPAttrType]bgp.PathAttributeInterface {
+	if len(attrs) == 0 {
```

Comment:
this seems quicker, but is it harmful to just return an empty map?
Why have this condition, just make the map, loop over zero parts and return the map.

Reply:
This will actually make a slight difference in BigQuery: if we return nil, they BigQuery will treat this as null; but if we returns an empty slice, bigquery will add an empty array there, which means null checks in queries will return different results. 

btw, I don't think you are looking at the latest commit. I am returning a slice of attributes after a few updates.

It looks like nil is the preferred style because it will not pass nil check: https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices

Reply:
oh! ok, this sounds ok to me :)